public/js/table-of-contents.js

```diff
@@ -0,0 +1,104 @@
+const MAX_TRIES = 20;
+
+/**
+ * Wait for an element to exist within the DOM, return that element
+ * Will reject if the attempts exceeds MAX_TRIES
+ * @param {string} selector
+ * @returns HTMLElement
+ */
+const waitForElement = (selector) =>
+  new Promise((resolve, reject) => {
+    let interval;
+    let tries = 0;
+
+    interval = setInterval(() => {
+      const element = document.body.querySelector(selector);
+      if (!element || element.length === 0) {
+        tries++;
+        if (tries > MAX_TRIES) {
+          clearInterval(interval);
+          reject('Could not find element');
+        }
+
+        return;
+      }
+
+      clearInterval(interval);
+      resolve(element);
+    }, 100);
+  });
+
+const createToc = async (config) => {
+  const tocElement = config.tocElement;
+  const titleElements = config.titleElements;
+
+  let tocElementDiv;
+  try {
```

Comment:
This `createToc` function is mostly unchanged from `pdd-web`, this `try...catch` block is an exception.